src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Binders/EnumTypeModelBinder.cs

```diff
@@ -0,0 +1,121 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.ComponentModel;
+using System.Runtime.ExceptionServices;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.Mvc.ModelBinding.Binders
+{
+    /// <summary>
+    /// <see cref="IModelBinder"/> implementation to bind models for types deriving from <see cref="System.Enum"/>.
+    /// </summary>
+    public class EnumTypeModelBinder : IModelBinder
+    {
+        private readonly MvcOptions _mvcOptions;
+
+        public EnumTypeModelBinder(MvcOptions mvcOptions)
+        {
+            _mvcOptions = mvcOptions;
+        }
+
+        /// <inheritdoc />
+        public Task BindModelAsync(ModelBindingContext bindingContext)
+        {
+            if (bindingContext == null)
+            {
+                throw new ArgumentNullException(nameof(bindingContext));
+            }
+
+            var valueProviderResult = bindingContext.ValueProvider.GetValue(bindingContext.ModelName);
+            if (valueProviderResult == ValueProviderResult.None)
+            {
+                // no entry
+                return Task.CompletedTask;
+            }
+
+            bindingContext.ModelState.SetModelValue(bindingContext.ModelName, valueProviderResult);
+            var modelType = bindingContext.ModelMetadata.UnderlyingOrModelType;
+            var stringValue = valueProviderResult.FirstValue;
+
+            try
+            {
+                object model;
+                if (string.IsNullOrWhiteSpace(stringValue))
+                {
+                    model = null;
+                }
+                else
+                {
+                    var typeConverter = TypeDescriptor.GetConverter(modelType);
```

Comment:
Could this be cached in the ctor? `TypeDescriptor.GetConverter` does a bunch of work.